### PR TITLE
Configure pre-commit CI

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Part of #10516 

### Summary of the issue:
As a developer, I would like automation tools to fix the lint of my code when I open a PR.
After #16936, we have applied pre-commit on all files. However, there are no automation checks for some of these checks (like whitespaces changes). Which means without running pre-commit automatically, the repository may have pre-commit regressions.

### Description of user facing changes
Improved documentation for pre-commit hooks and the PR contribution process.

### Description of development approach
Setup [Pre-commit CI](https://pre-commit.ci/). This is a CI/CD tool to automatically apply pre-commit to PRs.

### Testing strategy:
Confirm build succeeds.
Ensure failures from AppVeyor (e.g lint failures) are caught by pre-commit.

### Known issues with pull request:
None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Introduced a Continuous Integration (CI) configuration for pre-commit hooks, allowing for task skipping and submodule support.
	- Added a new section on "Status checks" in the README to display real-time build status badges.
	
- **Documentation**
	- Restructured contributing guidelines for pull requests, clarifying the submission process and CI/CD testing.
	- Enhanced instructions for setting up pre-commit hooks, including a new section on manually running hooks.

### CODE REVIEW

**Comment by seanbudd** [comment]
@LeonarddeR - I think we will have the same problem for scons checkPot too, I imagine we need to set up visual studio build tools before we can run these checks. I think we will have to exclude checkPot, and scons source for now. Perhaps also unit tests.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+  autofix_commit_msg: "Pre-commit autofix"
+
 repos:
 
 - repo: https://github.com/pre-commit/pre-commit-hooks
   rev: v4.6.0
   hooks:
     # Checks python syntax
     - id: check-ast
+      exclude: source/comInterfaces/_944DE083_8FB8_45CF_BCB7_C477ACB2F897_0_1_0.py
```
Comment (seanbudd) [inline .pre-commit-config.yaml:20]:
```suggestion
      # File uses encoding not supported in Linux
      exclude: source/comInterfaces/_944DE083_8FB8_45CF_BCB7_C477ACB2F897_0_1_0.py
```

File: readme.md [hunk 2]
```diff
@@ -25,3 +25,10 @@ Their contribution helps us maintain the security and integrity of our releases.
 ## Contributing to NVDA
 If you would like to contribute to NVDA, you can read more information in our [contributing guide](./.github/CONTRIBUTING.md).
 This includes information on reporting issues, triaging issues, testing, translating, contributing code/documentation and creating add-ons.
+
+## Status checks
+
+* Alpha build status: [![Alpha build status](https://ci.appveyor.com/api/projects/status/sqeer6p8lc80lvqe/branch/master?svg=true)](https://ci.appveyor.com/project/NVAccess/nvda/branch/master)
+* Beta build status: [![Beta build status](https://ci.appveyor.com/api/projects/status/sqeer6p8lc80lvqe/branch/beta?svg=true)](https://ci.appveyor.com/project/NVAccess/nvda/branch/beta)
+* Pre-commit status (master): [![pre-commit.ci status](https://results.pre-commit.ci/badge/github/nvaccess/nvda/master.svg)](https://results.pre-commit.ci/latest/github/nvaccess/nvda/master)
+* CodeQL security analysis status (master): [![CodeQL security analysis status](https://github.com/nvaccess/nvda/actions/workflows/github-code-scanning/codeql/badge.svg?branch=master)](https://github.com/nvaccess/nvda/actions/workflows/github-code-scanning/codeql?query=branch%3Amaster)
```
Comment (seanbudd) [inline readme.md:34]:
```suggestion
* AppVeyor
   * Alpha build status: [![view latest alpha builds](https://ci.appveyor.com/api/projects/status/sqeer6p8lc80lvqe/branch/master?svg=true)](https://ci.appveyor.com/project/NVAccess/nvda/branch/master)
   * Beta build status: [![view latest beta builds](https://ci.appveyor.com/api/projects/status/sqeer6p8lc80lvqe/branch/beta?svg=true)](https://ci.appveyor.com/project/NVAccess/nvda/branch/beta)
* Pre-commit status (master): [![view pre-commit.ci checks on master](https://results.pre-commit.ci/badge/github/nvaccess/nvda/master.svg)](https://results.pre-commit.ci/latest/github/nvaccess/nvda/master)
* CodeQL security analysis status (master): [![view CodeQL security analysis checks on master](https://github.com/nvaccess/nvda/actions/workflows/github-code-scanning/codeql/badge.svg?branch=master)](https://github.com/nvaccess/nvda/actions/workflows/github-code-scanning/codeql?query=branch%3Amaster)
```

**Review by michaelDCurran — APPROVED** [review]

File: readme.md [hunk 3]
```diff
@@ -25,3 +25,11 @@ Their contribution helps us maintain the security and integrity of our releases.
 ## Contributing to NVDA
 If you would like to contribute to NVDA, you can read more information in our [contributing guide](./.github/CONTRIBUTING.md).
 This includes information on reporting issues, triaging issues, testing, translating, contributing code/documentation and creating add-ons.
+
+## Status checks
+
+* AppVeyor
+   * Alpha build status: [![view latest alpha builds](https://ci.appveyor.com/api/projects/status/sqeer6p8lc80lvqe/branch/master?svg=true)](https://ci.appveyor.com/project/NVAccess/nvda/branch/master)
+   * Beta build status: [![view latest beta builds](https://ci.appveyor.com/api/projects/status/sqeer6p8lc80lvqe/branch/beta?svg=true)](https://ci.appveyor.com/project/NVAccess/nvda/branch/beta)
+* Pre-commit status (master): [![view pre-commit.ci checks on master](https://results.pre-commit.ci/badge/github/nvaccess/nvda/master.svg)](https://results.pre-commit.ci/latest/github/nvaccess/nvda/master)
```
Comment (seanbudd) [inline readme.md:34]:
```suggestion
* AppVeyor
   * Alpha build status: [![view latest alpha builds](https://ci.appveyor.com/api/projects/status/sqeer6p8lc80lvqe/branch/master?svg=true)](https://ci.appveyor.com/project/NVAccess/nvda/branch/master)
   * Beta build status: [![view latest beta builds](https://ci.appveyor.com/api/projects/status/sqeer6p8lc80lvqe/branch/beta?svg=true)](https://ci.appveyor.com/project/NVAccess/nvda/branch/beta)
* Pre-commit status (master): [![view pre-commit checks on master](https://results.pre-commit.ci/badge/github/nvaccess/nvda/master.svg)](https://results.pre-commit.ci/latest/github/nvaccess/nvda/master)
```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,12 +1,24 @@
...
 
+# https://pre-commit.ci/
+# Configuration for Continuous Integration service
+ci:
+  # Can't run Windows scripts on Linux
+  # These also do not apply fixes, only run tests
+  skip: [scons-source, checkPot, unitTest, licenseCheck]
+  submodules: true
+  autoupdate_commit_msg: "Pre-commit autoupdate"
+  autofix_commit_msg: "Pre-commit autofix"
```
Comment (LeonarddeR) [inline .pre-commit-config.yaml:11]:
```suggestion
  autoupdate_commit_msg: "Pre-commit automatic update"
  autofix_commit_msg: "Pre-commit automatic fix"
```

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,12 +1,24 @@
...
+
 repos:
 
 - repo: https://github.com/pre-commit/pre-commit-hooks
   rev: v4.6.0
   hooks:
     # Checks python syntax
     - id: check-ast
+      # File uses encoding not supported in Linux
```
Comment (LeonarddeR) [inline .pre-commit-config.yaml:20]: Ugh, I actually believe that comtypes should consider to fix this, enforcing UTF-8 for their generated modules. I filed https://github.com/enthought/comtypes/issues/592

File: .pre-commit-config.yaml (hunk 4 above)
Comment (seanbudd) [reply .pre-commit-config.yaml:11]: Could you open a small PR to fix this up. Maybe auto-fix and auto-update instead?

File: .pre-commit-config.yaml (hunk 5 above)
Comment (seanbudd) [reply .pre-commit-config.yaml:20]: Thanks agreed

Comment (seanbudd) [reply .pre-commit-config.yaml:20]: Perhaps we should also consider adding encoding checks to pre-commit?

**Pull request merged**